src/main/java/io/jenkins/plugins/folderauth/FolderBasedAuthorizationStrategy.java

```diff
@@ -55,6 +56,7 @@
     private final Set<AgentRole> agentRoles;
     private final Set<GlobalRole> globalRoles;
     private final Set<FolderRole> folderRoles;
+    private final HashMap<Class, Set> roleMap;
```

Comment:
It is not clear to me why it is needed, It is not really used AFAICT, and it would be persisted on the disk

Reply:
oh its used for convenience to store and retrieve the different sets based on its type as key. i reckon it wouldn't take a lot more memory since it simply references the existing sets?